java/org/apache/tomcat/util/digester/SetPropertiesRule.java

```diff
@@ -62,7 +59,7 @@ public void begin(String namespace, String theName, Attributes attributes)
             if ("".equals(name)) {
                 name = attributes.getQName(i);
             }
-            String value = attributes.getValue(i);
+            String value = attributes.getValue(i).intern();
```

Comment:
FYI: https://shipilev.net/jvm/anatomy-quarks/10-string-intern

> In almost every project we were taking care of, removing String.intern() from the hotpaths, or optionally replacing it with a handrolled deduplicator, was the very profitable performance optimization. Do not use String.intern() without thinking very hard about it, okay?

Reply:
So Aleksey's article was used to determine that de-dup could help and then implemented it in the exact way that Aleksey proved was terrible?

Reply:
For the relatively small number of strings concerned, we did not observe any performance issues and intern() was considered to be a reasonable approach. Happy to review contrary data, should any be provided.

Reply:
What was the observed benefit from using String.intern()?
My take away from Aleksey's article was that String.intern should effectively never be used.
If the only observed benefit was removal of long lived "duplicate" strings, then the [G1 string dedup](http://openjdk.java.net/jeps/192) already provides the majority of that benefit with none of the drawbacks of intern usage.

Reply:
See previous comments.